src/DepartmentService.Business/Department/FindDepartmentCommand.cs

```diff
@@ -1,241 +1,243 @@
+
+      return null;
+    }
+
+    public FindDepartmentsCommand(
+      IBaseFindFilterValidator baseFindValidator,
+      IDepartmentRepository repository,
+      IDepartmentInfoMapper departmentMapper,
+      IUserInfoMapper UserMapper,
+      IDepartmentUserInfoMapper departmentUserMapper,
+      IRequestClient<IGetUsersDataRequest> rcGetUsersData,
+      IRequestClient<IGetImagesRequest> rcGetImages,
+      IRequestClient<IGetPositionsRequest> rcGetPositions,
+      ILogger<FindDepartmentsCommand> logger,
+      IResponseCreator responseCreator)
+    {
+      _baseFindValidator = baseFindValidator;
+      _repository = repository;
+      _departmentMapper = departmentMapper;
+      _userMapper = UserMapper;
+      _departmentUserMapper = departmentUserMapper;
+      _rcGetUsersData = rcGetUsersData;
+      _rcGetImages = rcGetImages;
+      _rcGetPositions = rcGetPositions;
+      _logger = logger;
+      _responseCreator = responseCreator;
+    }
+
+    public async Task<FindResultResponse<DepartmentInfo>> ExecuteAsync(FindDepartmentFilter filter)
+    {
+      if (!_baseFindValidator.ValidateCustom(filter, out List<string> errors))
+      {
+        return _responseCreator.CreateFailureFindResponse<DepartmentInfo>(
+          HttpStatusCode.BadRequest, errors);
+      }
+
+      FindResultResponse<DepartmentInfo> response = new() { Body = new() };
+
+      (List<DbDepartment> dbDepartments, int totalCount) = await _repository.FindAsync(filter);
+
+      if (dbDepartments == null || !dbDepartments.Any())
+      {
+        return response;
+      }
+
+      Dictionary<Guid, Guid> departmentsDirectors =
+        dbDepartments
+          .SelectMany(d => d.Users.Where(u => u.Role == (int)DepartmentUserRole.Director))
+          .ToDictionary(d => d.DepartmentId, d => d.UserId);
+
+      List<UserData> usersData = await GetUsersDataAsync(
+        departmentsDirectors.Values.ToList(),
+        response.Errors);
+
+      List<ImageData> imagesData = await GetImagesAsync(
+        usersData?.Where(ud => ud.ImageId.HasValue)?.Select(ud => ud.ImageId.Value).ToList(),
+        response.Errors);
+
+      List<PositionData> positionsData = await GetPositionsAsync(
+        usersData?.Select(u => u.Id).ToList(),
+        response.Errors);
+
+      UserData userData = null;
+
+      foreach (DbDepartment dbDepartment in dbDepartments)
+      {
+        userData = departmentsDirectors.ContainsKey(dbDepartment.Id)
+          ? usersData.FirstOrDefault(u => u.Id == departmentsDirectors[dbDepartment.Id])
+          : null;
+
+        response.Body.Add(
+          _departmentMapper.Map(
+            dbDepartment,
+            userData == null ?
+              null :
+              _departmentUserMapper.Map(
+                _userMapper.Map(userData, imagesData.FirstOrDefault(i => i.ImageId == userData.ImageId)),
+                dbDepartment?.Users.FirstOrDefault(du => du.UserId == userData.Id),

```

Comment:
dbDepartment can not be null, please remove '?' dbDepartment.Users.FirstOrDefault(du => du.UserId == userData.Id)